packages/server/api/src/app/workers/helper/redis-rate-limiter.ts

```diff
@@ -0,0 +1,96 @@
+import { JobType, triggerTimeoutSandbox } from '@activepieces/server-shared'
+import { isNil } from '@activepieces/shared'
+import { DefaultJobOptions, Job, Queue, Worker } from 'bullmq'
+import dayjs from 'dayjs'
+
+import { createRedisClient } from '../../database/redis-connection'
+import { AddParams, JOB_PRIORITY } from '../queue/queue-manager'
+import { redisHandler } from '../redis/redis-handler'
+
+
+const EIGHT_MINUTES_IN_MILLISECONDS = 8 * 60 * 1000
+const defaultJobOptions: DefaultJobOptions = {
+    attempts: 5,
+    backoff: {
+        type: 'exponential',
+        delay: EIGHT_MINUTES_IN_MILLISECONDS,
+    },
+    removeOnComplete: true,
+}
+
+
+let worker: Worker | null = null
+
+let queue: Queue | null = null
+
+
+export const redisRateLimiter = {
+
+    async init(): Promise<void> {
+        if (isNil(queue)) {
+            queue = new Queue(
+                'rateLimitJobs',
+                {
+                    connection: createRedisClient(),
+                    defaultJobOptions,
+                },
+            )
+            await queue.waitUntilReady()
+        }
+        if (isNil(worker)) {
+            const lockDuration = dayjs.duration(triggerTimeoutSandbox, 'seconds').add(3, 'minutes').asMilliseconds()
+            worker = new Worker('rateLimitJobs', null, {
```

Comment:
the null can be a function directly and add the code that handle jobs here.

check the example of how the function is inside, you can figure out which queue it belongs to by looking at job.data.type

https://github.com/taskforcesh/bullmq#:~:text=Process%20the%20jobs%20in%20your%20workers%3A